### ISSUE TITLE
Mautic Huge Discrepancy & Terribly Inaccurate Page Hit Tracking

### ISSUE THREAD

**Issue by Paul979** [mautic/mautic#3396]
What type of report is this:

| Q  | A
| ---| ---
| Bug report? | V
| Feature request? | 
| Enhancement? | 

## Description:
Only a small portion of page hits that my contacts make are registered.

Because of this I can't make double opt-in work, because that relies on page hit tracking in Mautic, see: https://www.mautic.org/blog/integrator/creating-a-double-opt-in-email-campaign/

See screenshot for just one of the many examples:

![mautic-madness](https://cloud.githubusercontent.com/assets/25543761/22830141/b69f8f24-efa5-11e6-8103-ae1babf087a3.png)

## If a bug:

| Q   | A
| --- | ---
| Mautic version | 2.5.1
| PHP version | 7.0

### Steps to reproduce:
1. Create a double optin campaign as described here: https://www.mautic.org/blog/integrator/creating-a-double-opt-in-email-campaign/
2. Set up an alternative page hit/click tracking method besides Mautic
3. Compare the page hit tracking results and see that Mautic misses a huge chuck your contacts page hits in its reporting
 
### Log errors: 

_Please check for related errors in the latest log file in [mautic root]/app/log/ and/or the web server's logs and post them here. Be sure to remove sensitive information if applicable._

**Comment by merlin0** [comment]
Did you try and update to 2.6? I agree with you the tracking in 2.5.1 is Inaccurate, esp when you try to track downloads via a link. The campaign decision is simply not usefull because even though the link is visited to download the file, the campaign event is added to an other contact with the same ip.

**Comment by Paul979** [comment]
No I didn't try to update, I gave up on Mautic for now. Not just because of this issue. But because of the many more I previously encountered. This one just broke the straw for me, because it makes it impossible to double opt-in my users.

"even though the link is visited to download the file, the campaign event is added to an other contact with the same ip"

I'm not sure I understand what you're saying there, how can another contact have the same IP? All my contacts have different IPs.

**Comment by luizeof** [comment]
@Paul979  the better way to double optin your users is make an asset and verifying if user download this asset. Works fine.

**Comment by Paul979** [comment]
@luizeof Ok that's good to know, thanks for the tip. I could do that but the fact that Mautic just plain doesn't work in many areas will still keep me from using it again, until or if ever the bugs will be taken seriously and squatted out.

**Comment by merlin0** [comment]
@Paul979  Actually it doens't, like I said in my post, often if a user opens it for let's say an Iphone or other devices that have a strickt cookie policy tracks the mail open but makes a new lead with the page hit and the download, that stays anonymous.  This is the case since Mautic switched to cookie tracking.

**Comment by Paul979** [comment]
Ah ok I get it now, that makes sense. Is there a specific reason why there's no IP tracking so that we're able to track leads across devices? It seems like a big limitation.

And also, even though the links in my emails are Mautic tracking links, they don't report clicks accurately also in my email report. Shouldn't those tracking links work independent of cookie tracking?

Adam on the community forums also reported that contacts created via the API end up with no page hit tracking, here: https://www.mautic.org/community/index.php/6869-goodbye-mautic-huge-discrepancy-terribly-inaccurate-page-hit-tr/

**Comment by woakes070048** [comment]
I agree there should be IP tracking. Is there a feature request for this.

**Comment by MxyzptlkFishStix** [comment]
@woakes070048

It's at the bottom of Configuration, Landing Page Settings.

**Comment by Paul979** [comment]
@MxyzptlkFishStix what do you mean? I don't see any IP tracking settings there.

See screenshot:

![configuration mautic](https://cloud.githubusercontent.com/assets/25543761/22855654/30de813a-f084-11e6-8e5f-32383d98e5b8.png)

From what I understand it's already been established that the tracking pixel there only tracks via a cookie.

**Comment by MxyzptlkFishStix** [comment]
You must be on 2.5.1 or earlier.  It's in 2.6.0.

**Comment by Paul979** [comment]
Ah ok, thanks for the tip then. You're right, I'm still on 2.5.1, was having upgrade anxiety :)

**Comment by Paul979** [comment]
@luizeof 

Regarding "the better way to double optin your users is make an asset and verifying if user download this asset. Works fine."

I just tried this and this also doesn't work for me. I've set the opt-in confirmation to be the download of the asset, and since I did that no leads are getting confirmed anymore.
